### PR TITLE
feat: Adds Setup CLI Command to Configure GraphQL Trusted Documents

### PR DESCRIPTION
**Pull request by dthyresson**
Fixes https://github.com/redwoodjs/redwood/issues/9739

* updates Project TOML to enable trusted documents
* generates types/codegen to create the trustedDocuments store api side
* configures GraphQLHandler to use the trustedDocuments store
* updates docs to add the setup command and reference it in TD setup
* tests the handler transform function to check that toml and GraphQLHandler are updated properly (and left alone if needed)

TOML configuration tries to see if graphql or fragments are already configured and inserted the trustedDocuments setting accordingly. Perhaps not 100% foolproof, but defensive check provided.

### CODE REVIEW

File: packages/cli/src/commands/setup/graphql/graphql.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import terminalLink from 'terminal-link'
+
+export const command = 'graphql <feature>'
+export const description = 'Set up GraphQL feature support'
+export const builder = (yargs) =>
+  yargs
+    .commandDir('./features')
```
Comment (Tobbe) [inline packages/cli/src/commands/setup/graphql/graphql.js:7]: @jtoar was trying to get rid of our usage of `commandDir` (see https://github.com/redwoodjs/redwood/pull/9750), but ran into some non-trivial issues. Let's not add any new code that uses `commandDir`, to not make it even more difficult to get rid of in the future.

**Review by Tobbe — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/setup/graphql/features/trusted-docs.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          )
+        )
+        return
+      }
+
+      // add import
+      const project = new Project()
+      const graphQlSourceFile =
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        project.addSourceFileAtPathIfExists(graphQlSourcePath)!
```
[the remark this answers is not in the document]
Comment (dthyresson) [reply packages/cli/src/commands/setup/graphql/features/trusted-docs.ts:60]: Gotcha. I copied this code from some examples ZenStack maybe in their project -- just to see if works. Will address.

File: packages/cli/src/commands/setup/graphql/features/trusted-docs.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    [
+      {
+        title:
+          'Update Redwood Project Configuration to enable GraphQL Trusted Documents ...',
+        skip: () => false,
+        task: () => {
+          const redwoodTomlPath = getConfigPath()
+          const originalTomlContent = fs.readFileSync(redwoodTomlPath, 'utf-8')
+
+          const tomlToAppend = `[graphql]\n  trustedDocuments = true`
```
[the remark this answers is not in the document]
Comment (dthyresson) [reply packages/cli/src/commands/setup/graphql/features/trusted-docs.ts:155]: Good idea. Will safeguard from extraneous dupes.

File: packages/cli/src/commands/setup/graphql/graphql.js (hunk 1 above)
Comment (dthyresson) [reply packages/cli/src/commands/setup/graphql/graphql.js]: Done.

**Review by Tobbe — APPROVED** [review]

**Pull request merged**